xds/internal/balancer/clusterresolver/resource_resolver_dns.go

```diff
@@ -119,9 +145,15 @@ func (dr *dnsDiscoveryMechanism) resolveNow() {
 // after its `Close()` returns. Therefore, we can guarantee that no methods of
 // the topLevelResolver are invoked after we return from this method.
 func (dr *dnsDiscoveryMechanism) stop() {
-	if dr.dnsR != nil {
-		dr.dnsR.Close()
-	}
+	dr.serializerCancel()
+	done := make(chan struct{})
+	dr.serializer.Schedule(func(context.Context) {
+		if dr.dnsR != nil {
+			dr.dnsR.Close()
+		}
+		close(done)
+	})
```

Comment:
Can we somehow put this at the top of the queue? That way it doesn't have to query/run more DNS Resolver operations that we know we will throw away before close.

Reply:
This change has been deleted.